cmd/helm/repo_add.go

```diff
@@ -160,10 +160,10 @@ func (o *repoAddOptions) run(out io.Writer) error {
 	// 2. When the config is different require --force-update
 	if !o.forceUpdate && f.Has(o.name) {
 		existing := f.Get(o.name)
+		existing.URL = existing.URLWithTrailingSlash()
```

Comment:
Just remove these two lines and change line 166 to

```
if c.URLWithTrailingSlash() != *existing.URLWithTrailingSlash() {
```

Like I mentioned back in December with https://github.com/helm/helm/pull/9095#discussion_r536440127. Please ignore [this comment suggestion](https://github.com/helm/helm/pull/9095#discussion_r536517277) - by doing what he suggested, you are overwriting a variable which can affect downstream behaviour. He even called that out in another comment further down: https://github.com/helm/helm/pull/9095#discussion_r536802134